tabular/src/autogluon/tabular/models/fastainn/tabular_nn_fastai.py

```diff
@@ -377,7 +379,18 @@ def __get_objective_func_to_monitor(self, objective_func_name):
             objective_func_name_to_monitor = monitor_obj_func[objective_func_name]
         return objective_func_name_to_monitor
 
+    # FIXME: torch.set_num_threads(self._num_cpus_infer) is required because XGBoost<=1.5 mutates global OpenMP thread limit
+    #  If this isn't here, inference speed is slowed down massively.
+    #  Remove once upgraded to XGBoost>=1.6
     def _predict_proba(self, X, **kwargs):
+        import torch
+        num_threads = torch.get_num_threads()
+        torch.set_num_threads(self._num_cpus_infer)
+        pred_proba = self._predict_proba_internal(X=X, **kwargs)
+        torch.set_num_threads(num_threads)
+        return pred_proba
```

Comment:
can we wrap thread manipulation logic to do this via context and reuse the logic here and below?
```python
with torch_thread_fix():
         pred_proba = self._predict_proba_internal(X=X, **kwargs)
return pred_proba
```
Or.. maybe mixin so the code is reused vs copy-pasted

Reply:
added context